src/datasources/human-description-api/human-description-api.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IHumanDescriptionApi } from '../../domain/interfaces/human-description-api.interface';
+import {
+  MessageTemplates,
+  HumanReadableFragment,
+  ValueType,
+  Expression,
+} from './entities/human-description.entity';
+
+const isValueType = (type: unknown): type is ValueType => {
+  return Object.values(ValueType).includes(type as ValueType);
+};
+
+@Injectable()
+export class HumanDescriptionApi implements IHumanDescriptionApi {
+  private readonly parsedMessages: MessageTemplates;
+
+  /**
+   *  Regex Template that matches two patterns
+   *  1. Double curly braces consisting of 2 groups separated by a space
+   *  2. Any non-whitespace character i.e. simple words
+   */
+  private readonly TEMPLATE_REGEX = /{{(.*?)\s(\$.*?)}}|(\S+)/g;
+
+  constructor(
+    @Inject('ContractDescriptions')
+    private readonly contractDescriptions: Expression,
+  ) {
+    this.parsedMessages = this.parseMessages(this.contractDescriptions);
+  }
+
+  getParsedMessages(): MessageTemplates {
+    return this.parsedMessages;
+  }
+
+  parseMessages(messages: Expression): MessageTemplates {
```

Comment:
Is there a way to reduce the cyclomatic complexity of this function (break it in smaller functions, etc.) to make it more readable? It contains several loops and nested conditionals, it took me a while to think whether the `while` loop really has confiable exit point and it won't let to infinite loops no matter the input 🙂